pkg/backend/identityapi/register_client_test.go

```diff
@@ -168,3 +177,120 @@ func TestRegisterMarshallCompression(t *testing.T) {
 	assert.NoError(t, err)
 	assert.EqualValues(t, mtInput, mtOutput)
 }
+
+func TestRegisterClient_RegisterEntity(t *testing.T) {
+	mc := &mockAPIClient{}
+	entID := rand.Int63n(10000)
+	expectedRegisterRequest := identity.RegisterRequest{
```

Comment:
why not using `newRegisterRequest` constructor instead?

Reply:
If anything wrong with `newRegisterRequest` it will be picked up by this test but wouldn't fail if I used that function. I could add a test around `newRegisterRequest` but I feel that is just internals of the function `RegisterEntity` and the above test covers it. There is definitely room for refactoring in the test (like the setup we do for some scenarios) which probably should be moved into a table driven test.